src/imitation/algorithms/preference_comparisons.py

```diff
@@ -499,17 +541,21 @@ def train(self, iterations: int):
         Args:
             iterations: number of iterations of the outer training loop
         """
-        for _ in range(iterations):
+        for i in range(iterations):
             self.logger.log(f"Collecting {self.sample_steps} trajectory steps")
             trajectories = self.trajectory_generator.sample(self.sample_steps)
             self._check_fixed_horizon(trajectories)
             self.logger.log("Creating fragment pairs")
             fragments = self.fragmenter(trajectories)
-            self.logger.log("Gathering preferences")
-            preferences = self.preference_gatherer(fragments)
+            with self.logger.accumulate_means("preferences"):
+                self.logger.log("Gathering preferences")
+                preferences = self.preference_gatherer(fragments)
             self.dataset.push(fragments, preferences)
             self.logger.log(f"Dataset now contains {len(self.dataset)} samples")
-            self.logger.log("Training reward model")
-            self.reward_trainer.train(self.dataset)
-            self.logger.log(f"Training agent for {self.agent_steps} steps")
-            self.trajectory_generator.train(steps=self.agent_steps)
+            with self.logger.accumulate_means("reward"):
+                self.logger.log("Training reward model")
+                self.reward_trainer.train(self.dataset)
+            with self.logger.accumulate_means("agent"):
+                self.logger.log(f"Training agent for {self.agent_steps} steps")
+                self.trajectory_generator.train(steps=self.agent_steps)
+            self.logger.dump(i)
```

Comment:
FYI I think this is meant to be globally unique, so weird things may happen with logging if you call `train` multiple times (but I think we don't do that?)

Reply:
Our script only calls it once, but I think we should support calling it multiple times (as long as it's not much work), so I've added a global step counter